lib/rubocop/cop/rails/relative_date_constant.rb

```diff
@@ -25,6 +25,8 @@ class RelativeDateConstant < Cop
         RELATIVE_DATE_METHODS = %i[ago from_now since until].freeze
 
         def on_casgn(node)
+          return if node.children.last.lambda_or_proc?
```

Comment:
Btw, it seems to me you can return quickly for anything but a `send` node on the right-hand side of the assignment. Obviously - if it's not a `send` it can't be what this cop is checking for. 

Overall this cop is implemented in a rather odd manner and can benefit from using node patterns IMO.